x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/RestEsqlTestCase.java

```diff
@@ -440,6 +440,38 @@ public void testOutOfRangeComparisons() throws IOException {
         }
     }
 
+    public void testInternalRange() throws IOException {
+        final int NUM_SINGLE_VALUE_ROWS = 100;
+        bulkLoadTestData(NUM_SINGLE_VALUE_ROWS);
+
+        String upperBound = randomFrom(" < ", " <= ");
+        String lowerBound = randomFrom(" > ", " >= ");
+
+        String predicate = "{}" + upperBound + "{} and {}" + lowerBound + "{} and {} != {}";
+
+        List<String> predicates = List.of(
+            format(null, predicate, "integer", NUM_SINGLE_VALUE_ROWS + 1, "integer", -1, "integer", -1),
+            format(null, predicate, "long", NUM_SINGLE_VALUE_ROWS + 1, "long", -1, "long", -1),
+            format(null, predicate, "double", NUM_SINGLE_VALUE_ROWS + 1.0, "double", -1.0, "double", -1),
+            format(null, predicate, "date", "\"9999-12-31\"", "date", "\"1001-01-01\"", "date", "\"1000-01-01\""),
+            format(null, predicate, "keyword", "\"l\"", "keyword", "\"j\"", "keyword", "\"a\""),
+            format(null, predicate, "ip", "\"128.0.0.0\"", "ip", "\"126.0.0.0\"", "ip", "\"129.0.0.0\""),
+            format(null, predicate, "version", "\"2\"", "version", "\"1\"", "version", "\"0\"")
```

Comment:
nit: there's other data types we could easily include, like byte, short, scaled float, half float, float, unsigned long. C.f. [here](https://www.elastic.co/guide/en/elasticsearch/reference/master/esql-limitations.html) for the (officially) supported types.

Reply:
More types are added.